### ISSUE TITLE
search bar doens't work and it keeps loading forever 

### ISSUE THREAD

**Issue by johnwise9** [Grape-Labs/grape-art#5]
the program keeps loading forever, and search for address doesn't work most of the time. the same with the selling tap, following tap ... just keeps loading

**Comment by johnwise9** [comment]
<img width="1226" alt="Screenshot 2022-03-28 at 22 45 18" src="https://user-images.githubusercontent.com/97042650/160492581-1e3b4f73-7760-4817-b19d-07b4a24c1542.png">

**Comment by kirksgithub** [comment]
Hi John, can you share that public key here so I can give it a spin on my end to see what is causing this

**Comment by johnwise9** [comment]
<img width="1265" alt="Screenshot 2022-03-28 at 22 47 01" src="https://user-images.githubusercontent.com/97042650/160492778-bfdd42ef-6087-4626-96c7-3a9f2a7caebf.png">

**Comment by johnwise9** [comment]
I just built the same version and pushed it with exact same settings, so the same pup key in the Repo. just wanted to check it out

**Comment by kirksgithub** [comment]
To validate if a wallet address is valid, grape.art checks if the public key is on curve https://solana-labs.github.io/solana-web3.js/classes/PublicKey.html#isOnCurve - the address you are trying is not on curve, can you test GrapevviL94JZRiZwn2LjpWtmDacXU8QhAJvzpUMMFdL which is a valid wallet address

**Comment by johnwise9** [comment]
storm9aw7D6eDEdTo65187oGSzqnWCmJcvhf17FHW7H, 

the thing is that on grape.art it works super smooth and fast. but the repo it's unfortunately not functional, it works at times for a mint or so and then would start loading for ever

**Comment by johnwise9** [comment]
ok got it, I sreached for GrapevviL94JZRiZwn2LjpWtmDacXU8QhAJvzpUMMFdL and it worked. but still the feed is reloading

**Comment by johnwise9** [comment]
the repo is super awesome, will you add in the future searching for collections as well not only addresses?

**Comment by johnwise9** [comment]
for example AjwiSEA8CHfJ2TnGJRRUHmHVtHhTHWTGdfGYSv4bjP4o works fine when try it on grape.art 
however on this Repo it does't work 


<img width="1465" alt="Screenshot 2022-03-28 at 23 06 59" src="https://user-images.githubusercontent.com/97042650/160495325-2b5e42b1-25b3-4d98-846a-c3e1ed70060b.png">
<img width="1389" alt="Screenshot 2022-03-28 at 23 07 08" src="https://user-images.githubusercontent.com/97042650/160495338-1629aaaa-e2db-46d1-b6a6-82c5d43084fd.png">

**Comment by kirksgithub** [comment]
Thank you so much for your feedback! This was created as proof of concept, our goal was to build something that was true to the Solana ethos and now we are focused on collections/groups/curation (yes the goal is that you will be able to navigate and search collections)
Grape.art is decentralized, so RPC calls are quite heavy but there is absolutely room for improvement, you may consider using an alternative RPC provider (you can quickly plug that in at utils/grapeTools/contants.tsx)

And if you would like to help contribute on improvements feel free to do so ;)

As for the address, I did check AjwiSEA8CHfJ2TnGJRRUHmHVtHhTHWTGdfGYSv4bjP4o I actually do see an issue, we are using the Solana CDN to boost the loading of the arweave images, I think we need a fallback there when it fails to simply load the arweave file

**Comment by kirksgithub** [comment]
@johnwise9 thank you for your feedback, this issue regarding the loading images has been updated on this build, we now do an async call to verify if they are indeed loading (GalleryItem.tsx line 111)

**Comment by johnwise9** [comment]
thanks, awesome man you're doing 

amazing job, and yes would love to help out and contribute as well. I just checked now, it works, but you can't list something, when I try to list, transaction doesn't go through : 

<img width="1661" alt="Screenshot 2022-03-30 at 16 56 16" src="https://user-images.githubusercontent.com/97042650/160879797-1ea08a2b-864e-4dab-a098-fe1031da35ac.png">

**Comment by kirksgithub** [comment]
This seems to be because Solana is running slow, this happens some times during the week/or during the day when there is a bottle neck of requests. Transaction is cannot be confirmed on the network due to latency, so first check the sig (or just refresh the page to see if it is listed)

**Comment by kirksgithub** [comment]
Yeah mine was listed just took too long to confirm the tx too (this is a max timeout of 30s to confirm the transaction)

**Comment by johnwise9** [comment]
Yes, I tried it again, was the same. let me try again. is it the same with a custom RPC, as I'm running it with a custom RPC and still the same.

**Comment by kirksgithub** [comment]
for transactions best you use the Solana endpoint https://docs.solana.com/cluster/rpc-endpoints

**Comment by johnwise9** [comment]
ok so https://solana-api.projectserum.com is better off than using a custom RPC for transactions?

**Comment by johnwise9** [comment]
it's the same unfortunately, I've tried a custom RPC, I've tried Solana one https://api.mainnet-beta.solana.com, and it's the same  :(
